flask_sqlalchemy/__init__.py

```diff
@@ -515,9 +515,14 @@ def __get__(self, obj, type):
         try:
             mapper = orm.class_mapper(type)
             if mapper:
-                return type.query_class(mapper, session=self.sa.session())
+                query = type.query_class(mapper, session=self.sa.session())
         except UnmappedClassError:
-            return None
+            query = None
+
+        if obj is not None and not obj.__class__.__bases__[0].query:
+            raise RuntimeError('Only use BaseQuery helper from a model class.')
```

Comment:
Mention that the error is because they're accessing from an instance: "The query property is only accessible from a model class, not an instance."